### PR TITLE
Validate deployment

### PR DESCRIPTION
**Pull request by louischan-oursky**

### CODE REVIEW

File: src/api/deployment.ts [hunk 1]
```diff
@@ -10,6 +10,31 @@ import {
 } from '../types';
 import { callAPI } from './skygear';
 
+export async function validateDeployment(
+  context: CLIContext,
+  deployments: { [name: string]: DeploymentItemConfig },
+  hooks: HookConfig[]
+): Promise<void> {
+  const deploymentsRequestPayload: {
+    [key: string]: ReturnType<
```
Comment (kiootic) [inline src/api/deployment.ts]: Is it intentional that type `DeploymentItemConfigPayload` is not used?

File: src/api/skygear.ts [hunk 2]
```diff
@@ -15,24 +14,22 @@ function defaultHeaders(context: CLIContext) {
...
-      return p;
-    })
-    .catch((_error) => {
-      throw new Error(response.statusText);
-    });
+  // By default, we use statusText as error message
+  // This is a fallback only.
+  const preparedError = new Error(response.statusText);
+  // Attach response to error
+  (preparedError as any).response = response;
```
Comment (kiootic) [inline src/api/skygear.ts]: Should we have a subclass of `Error` for controller network errors? It'll avoid the need of duck typing to handle our errors.

File: src/error.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+export function printError(error: any) {
+  let s = '';
+
+  // skyerr
+  if (error && error.json && error.json.error) {
+    const skyerr = error.json.error;
+    switch (skyerr.code) {
+      // invalid argument
+      case 108:
```
Comment (kiootic) [inline src/error.ts]: I think we should have an enum (or some constant declarations if you prefer) for possible error codes.

File: src/api/deployment.ts (hunk 1 above)
Comment (louischan-oursky) [reply src/api/deployment.ts]: The reason I can think of is that `validateDeployment` is copied from `createDeployment` and `createDeployment` is older than `DeploymentItemConfigPayload`. I will update both to use `DeploymentItemConfigPayload`. Good catch!

File: src/error.ts (hunk 3 above)
Comment (louischan-oursky) [reply src/error.ts]: Actually the full list of codes is defined in the skygear SDK. Should we duplicate that list here? Or we just add the code as we need it? Anyway I agree that we should introduce named constant for error code.

File: src/api/skygear.ts (hunk 2 above)
Comment (louischan-oursky) [reply src/api/skygear.ts]: Let me introduce a `HTTPError`

**Comment by louischan-oursky** [comment]
@kiootic Comments addressed.

**Review by kiootic — APPROVED** [review]

**Pull request merged**